bgpd/bgp_clist.c

```diff
@@ -981,6 +1028,27 @@ bool lcommunity_list_any_match(struct lcommunity *lcom,
 	return false;
 }
 
+bool ecommunity_list_any_match(struct ecommunity *ecom, struct community_list *list)
+{
+	struct community_entry *entry;
+	uint8_t *ptr;
+	uint32_t i;
+
+	for (i = 0; i < ecom->size; i++) {
+		ptr = ecom->val + (i * ECOMMUNITY_SIZE);
```

Comment:
We should handle IPv6-specific extended communities also. Thus ecom->unit_size should be used instead.

Reply:
ok